src/main/java/com/dev/cinema/validators/EmailValidator.java

```diff
@@ -0,0 +1,19 @@
+package com.dev.cinema.validators;
+
+import com.dev.cinema.annotations.ValidEmail;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+public class EmailValidator implements ConstraintValidator<ValidEmail, String> {
+    public static final Pattern VALID_EMAIL_ADDRESS_REGEX =
+            Pattern.compile("^[A-Z0-9._%+-]+@[A-Z0-9.-]+\\.[A-Z]{2,6}$",
+                    Pattern.CASE_INSENSITIVE);
+
+    @Override
+    public boolean isValid(String email, ConstraintValidatorContext constraint) {
+        Matcher matcher = VALID_EMAIL_ADDRESS_REGEX.matcher(email);
```

Comment:
I think you should add a check for a `null` email here or in the DTO

Reply:
let's add the null check for email here as well